docs/contributing/docs-and-website-components.md

```diff
@@ -0,0 +1,409 @@
+---
+title: Docs and Website Components
+tableOfContentsDepth: 2
+---
+
+import LayerModel from "../../www/src/components/layer-model"
+import HorizontalNavList from "../../www/src/components/horizontal-nav-list"
+import Breadcrumb from "../../www/src/components/docs-breadcrumb"
+import { itemListContributing } from "../../www/src/utils/sidebar/item-list"
+import TableOfContents from "../../www/src/components/docs-table-of-contents"
+
+<div onClick={console.log({ itemListContributing, props })} />
+
+The Gatsbyjs.org site has a handful of components that have already been developed to facilitate writing new content for the blog and the docs. There are also components that are good to be aware that help organize and lay out content in various pages.
+
+This guide lists what components are available and explains how to use them. You can also refer to the [code for this page](https://github.com/gatsbyjs/gatsby/blob/master/docs/contributing/docs-and-website-components.md) to see to how each component can be used, because each is embedded on this page!
+
+Information about authoring in markdown and styling components on the site is also listed.
+
+## Globally available components
+
+A variety of components have been written to help with formatting code and content across the blog and the docs and don't need to be imported.
+
+---
+
+### Guide list
+
+The `<GuideList />` is a component that renders a list of docs nested below the current doc. It is often used on overview pages like the [Headless CMS](/docs/headless-cms/) page where many other pages are nested below it, to show what a doc contains.
+
+#### Usage
+
+It takes one prop:
+
+- `slug` - the value of which is already available on every page by default
```

Comment:
```suggestion
- `slug` - the value of which is already available on every page's context on Gatsbyjs.org by default
```
Adding extra explicitness in case people want to copy/port these components, so they know there's more involved than just the component file itself.